### ISSUE TITLE
Tool supplies drop boxes 1-3 hours away from entered address, instead of much closer drop box.

### ISSUE THREAD

**Issue by MariaBianchi** [votinginfoproject/vip-embeddable-tool#167]
The state of Oregon let us know: 

As it relates to our discussion, the following is the address that was directing to locations quite far away, the first time I entered it the tool displayed a location very close (Umatilla City Hall), the next time it came up with a library near Portland (Ledding Library of Milwaukie)(~3 hours away), and another time it suggested Baker County Clerk's Office (~1.5 hours away).
 
69958 Schiller Dr.
Echo OR 97826

Do you know why the tool wouldn't be suggesting the closest drop box in this case?

**Comment by kthale** [comment]
@alexkonrad bumping this--can you please look into this today, if possible (or email me and let me know if it's not). 

Thanks so much!

**Comment by alexkonrad** [comment]
@kthale is there an EID for this for testing?

**Comment by acohen84** [comment]
@alexkonrad Oregon is 4165.  Hope that helps!

**Comment by acohen84** [comment]
@alexkonrad Any update?

**Comment by alexkonrad** [comment]
I'm still looking into this -- I've noticed once the Milwaukie Library switched with Umatilla City Hall. Because of API limitations the tool isn't able to tell how far a location is from the inputted address based on the response, but we've been returning locations in the order returned by the Google API (which is in order of proximity). I can't find any evidence that it could be switching location order yet, but I'll go through the code and see if I can find anything there.

**Comment by acohen84** [comment]
@alexkonrad Hi Alex, just checking in on this, because we're seeing it in other states.  Any update?

**Comment by alexkonrad** [comment]
@acohen84 can you give me another example? Thanks

**Comment by acohen84** [comment]
@alexkonrad Sure: 8156 Oliver Rd, Peyton, CO.  The state flagged this for us, and in looking myself, the Early Vote site is showing up in Virginia (!).  The state noted that every time they searched they were getting a different location.

**Comment by alexkonrad** [comment]
@acohen84 Sorry to keep asking for this: do you have an EID for that CO address? I don't see any Colorado elections in the list right now. (In general, is there a place I can look up the active EIDs when you guys give me an address?)

I'm going to keep looking to see if there's anything I can do on my end, but right now it's looking like this is just the order in which data is returned by the Google API. For example, when I use their [testing widget](https://developers.google.com/civic-information/docs/v2/elections/voterInfoQuery) with the OR address and EID 4165 I get varying results for the first drop-off location:

![image](https://cloud.githubusercontent.com/assets/4000470/10755835/91f4b2c4-7c76-11e5-8443-78bbacb31a8d.png)
![image](https://cloud.githubusercontent.com/assets/4000470/10756169/b3e27e78-7c78-11e5-989a-e38b7a1fbe9f.png)
![image](https://cloud.githubusercontent.com/assets/4000470/10756185/c9f1fe0a-7c78-11e5-939f-e67943a26add.png)

If there's a bug in the tool where it's reordering locations I can fix that, but if the API is not returning drop-off locations in the right order then I can't reorder them inside the tool: that would involve looking up the geocodes for every single location before even displaying the first.

When we were building the tool there was a discussion about whether the API returns locations in order of proximity, and I believe Jared was able to get confirmation from Google on this. Is this limited to just drop-off locations or are there addresses where this happens with other types of polling locations?

**Comment by kthale** [comment]
@alexkonrad you can find the EIDs here: https://github.com/votinginfoproject/VIP-feed-information-project/blob/master/datafeed.txt

**Comment by MariaDemocracyWorks** [comment]
@alexkonrad I checked with Google, and according to Chetan, the API should be supplying results in order of proximity, but are not for the Echo address because they can't geocode a lat/long.

The same isn't true for the CO address, but I'm not seeing that bug consistently duplicated, either.

**Comment by acohen84** [comment]
@alexkonrad Thanks for looking into the Oregon address.  Now that that's resolved - 

- In a state like Colorado that has multiple modes of voting, how do we prioritize the voting location that is returned to the voter?  It seems like the first option someone sees in the tool is always coded as a polling location, even if that's not the closest option to the address they entered.  This is the case with 8156 Oliver Rd, Peyton, CO, where the North Branch County Clerk's office displays first and most prominently, but there is a dropoff location closer to the Oliver Road address.

- In searching the above address, every point on the map is showing up with the North Branch County Clerk's Office name, address, and hours.  This was happening yesterday, and I've done several hard refreshes and it's still happening.  This is not happening for Maria.  I'm using Mozilla.  Is this a problem with the tool or with me?

**Comment by MariaDemocracyWorks** [comment]
More fun with Colorado: I spoke to the elections official again, and she clarified the error report she'd been making. When she clicks on the grey drop box icon in the key, it takes her to a dropbox, though never the closest one. Are the icons in the key intended to be a form of navigation?

Her exact words were "When I did multiple searches on 8168 Oliver Rd (right next door to my address), it never showed the same drop-off location first, and when I click the drop-off icon multiple times, it changes locations, but never shows the Falcon location. Once it chooses the Calhan location, clicking the icon no longer results in a different location being shown."

**Comment by alexkonrad** [comment]
@acohen84 to your two points:

- Unfortunately there's no way for us currently to determine whether the closest polling location is closer to the user's registered address than a drop-off location or an early vote location. This is because they're all returned as separate groups of locations by the Google API, each of which is sorted by order of proximity. There's no way for us to compare the proximity of different types of locations before we geocode them. I imagine there could be a possible solution which would geocode series of locations in parallel and then determine proximity by comparison, but there might be tradeoffs to this approach and it would likely involve additional scope.

- Thanks for letting me know, this is probably a Firefox-specific bug. You can open a separate issue for this, or I will when I can reproduce it on Firefox.

@MariaDemocracyWorks:

- Thanks for sending me that response from Colorado. The click on legend icon is supposed to be a feature that takes you to the closest voting location of that type. Let me look into this more, but at this point I'm assuming it's having difficulty based on the number of locations. I could temporarily disable this feature while we decide on a solution for this.

- The location legend click issue seems like a different issue than searching for 8168 Oliver Rd and being taken to Virginia. I haven't been able to replicate this myself yet on election 4147 -- is this happening repeatedly?

**Comment by MariaDemocracyWorks** [comment]
@alexkonrad To clarify about ordering, but it seems to come in a variety of flavors. I'm going to go ahead and start new issues, since this thread has become pretty convoluted, but I appreciate the explanation of the click on legend feature.

I really don't understand what's happening with Amy's map to Virginia bug, but I also haven't been able to replicate that one, and haven't gotten any similar reports from users.
